src/Databases/MySQL/MaterializeMySQLSyncThread.cpp

```diff
@@ -214,10 +215,32 @@ void MaterializeMySQLSyncThread::stopSynchronization()
 
 void MaterializeMySQLSyncThread::startSynchronization()
 {
-    const auto & mysql_server_version = checkVariableAndGetVersion(pool.get());
+    try{
+        const auto & mysql_server_version = checkVariableAndGetVersion(pool.get());
 
-    background_thread_pool = std::make_unique<ThreadFromGlobalPool>(
-        [this, mysql_server_version = mysql_server_version]() { synchronization(mysql_server_version); });
+        background_thread_pool = std::make_unique<ThreadFromGlobalPool>(
+            [this, mysql_server_version = mysql_server_version]() { synchronization(mysql_server_version); });
+    }
+    catch (...)
+    {
+        try
+        {
+            throw;
```

Comment:
Why do we need to rethrow it?

Reply:
To devide the exception class. e.g. the exception is mysqlconnectionerr, the user accessed denied. And I want to use 

```
try
{}
catch (mysqlxx::ConnectionFailed & e)
{}
catch (...)
{throw;}

```

But I see the other code in CH is 

```
try
{ ...; }
catch (...)
{
catch (A & E)
{
}
catch (B & E)
{
}
}
```

So I used this method.